### PR TITLE
Users api integration

### PR DESCRIPTION
**Pull request by pipeache**
## Summary

- Service and endpoint for user creation implemented

## Known Issues

None

## Trello Card

https://trello.com/c/jJWAqEgf/35-sign-up

### CODE REVIEW

File: app/models/user.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
+module.exports = (sequelize, DataTypes) => {
+  const User = sequelize.define(
+    'User',
+    {
+      first_name: { type: DataTypes.STRING, allowNull: false },
+      last_name: { type: DataTypes.STRING, allowNull: false },
+      email: { type: DataTypes.STRING, allowNull: false, unique: true },
+      password: { type: DataTypes.STRING, allowNull: false }
```
Comment (dbenavidesv) [inline app/models/user.js]: Model properties should be on `camelCase`

File: app/controllers/users.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
+const logger = require('../logger');
+
+const { registerUser } = require('../services/users');
+
+exports.createUser = async (req, res) => {
+  const { first_name, last_name, email, password } = req.body;
```
Comment (dbenavidesv) [inline app/controllers/users.js]: Use a mapper to convert the `snake_case` input properties to `camelCase`

File: app/controllers/users.js [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+      hasErrors: false,
+      message
+    });
+  } catch (error) {
+    logger.error(error);
+
+    res.status(422).json({
+      hasErrors: true,
+      message: error.message
+    });
```
Comment (dbenavidesv) [inline app/controllers/users.js]: There is an error handler middleware, use `next(error)` here to utilize that middleware. And check the `errors.js` file to create custom errors

File: app/middlewares/users.js [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+  try {
+    isValidDomainEmail(email);
+    isValidPassword(password);
+
+    next();
+  } catch (error) {
+    res.status(400).json({
+      hasErrors: true,
+      message: error.message
+    });
```
Comment (dbenavidesv) [inline app/middlewares/users.js]: Same here, use `next(error)`

File: app/routes.js [hunk 5]
```diff
@@ -1,9 +1,11 @@
-// const controller = require('./controllers/controller');
+const userController = require('./controllers/users');
 const { healthCheck } = require('./controllers/healthCheck');
 
+const { registerUser } = require('./middlewares/users');
+
 exports.init = app => {
   app.get('/health', healthCheck);
-  // app.get('/endpoint/get/path', [], controller.methodGET);
```
Comment (dbenavidesv) [inline app/routes.js:6]: Remove all the commented example lines

File: migrations/migrations/20210628172053-create-table-users.js [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+  up: (queryInterface, Sequelize) =>
+    queryInterface.createTable('users', {
+      id: {
+        type: Sequelize.INTEGER,
+        primaryKey: true,
+        autoIncrement: true
+      },
+      first_name: {
+        type: Sequelize.STRING,
+        defaultValue: false,
```
Comment (dbenavidesv) [inline migrations/migrations/20210628172053-create-table-users.js]: There's no need to add this

Comment (pipeache) [reply migrations/migrations/20210628172053-create-table-users.js]: Does the Sequelize data type default to String ? or do you mean `defaultValue: false` ?

File: app/controllers/users.js (hunk 2 above)
Comment (pipeache) [reply app/controllers/users.js]: Solved, thanks

File: app/models/user.js (hunk 1 above)
Comment (pipeache) [reply app/models/user.js]: I made changes in the definition of the model, I did not know the field attribute to match with what was defined in the table. Solved, thanks

File: migrations/migrations/20210628172053-create-table-users.js (hunk 6 above)
Comment (dbenavidesv) [reply migrations/migrations/20210628172053-create-table-users.js]: I mean `defaultValue = false`

Comment (pipeache) [reply migrations/migrations/20210628172053-create-table-users.js]: Solved, thanks!

**Review by dbenavidesv — APPROVED** [review]

**Pull request merged**